### ISSUE TITLE
Provide a Mojo to export the 'Eclipse-SourceBundle' value as property

### ISSUE THREAD

**Issue by HannesWell** [eclipse-tycho/tycho#1795]
The `tycho-source-plugin` provides the goal `plugin-source` that adds the `Eclipse-SourceBundle` header with a suitable value to a existing `sources` artifact. It does this by creating a copy of the original jar where that header+value is added to the Manifest.

I think it would good to provide another mojo that just exports the header's value as a property created during the build.
One can then use that value for example in the `bnd` configuration section of the `bnd-maven-plugin` to add the header in the following way:
```
Eclipse-SourceBundle: ${tycho.eclipse.source.bundle}
```
This would save the creation of a second jar (and thus slightly improve the performance) and would align better with the mojo proposed in https://github.com/eclipse-tycho/tycho/issues/1794 and would make it more clear to the user what is happening (IMHO).

**Comment by laeubi** [comment]
Exporting it as a property seems not right, so maybe you want explain what is the problem with copy the existing one? If one likely the header can be generated by BND itself, but actually it would be much better to embedd the source in the bundle itself with `OSGi-OPT` folder.... `Eclipse-SourceBundle` is really a hack that better dies earlier than later instead of carry this to other eco-systems....

**Comment by HannesWell** [comment]
> but actually it would be much better to embedd the source in the bundle itself with `OSGi-OPT` folder.... `Eclipse-SourceBundle` is really a hack that better dies earlier than later instead of carry this to other eco-systems....

I don't think it is a good idea to generally put sources into binary artifacts. From my experience that approximately doubles the size of the jar, which is really bad if you are not interested in the sources. And I don't think that most people are interested of the sources of their Eclipse IDE and would be happy about a doubled download size.
Maven also by default splits the main and the sources artifacts so in general this is no Eclipse hack or not special to Eclipse. The technical implementation with the header might be a Eclipse specialty. But because the p2-repo layout is flat and not as structured as the maven-repo layout, I think it is good to have some metadata to explicitly associating a main artifact and its source instead of guessing based on some name rules or alike.
